File: solutions/csharp/sum-of-multiples/1/SumOfMultiples.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        var points = new List<int>();
        foreach (int item in multiples)
        {
            if (item == 0)
            {
                continue;
            }
            for (int i = item; i < max; i += item)
            {
                points.Add(i);
            }
        }
        return points.Distinct().Sum();
    }
}

File: solutions/fsharp/sum-of-multiples/1/SumOfMultiples.fs
module SumOfMultiples

let sum (numbers: int list) (upperBound: int): int = 
    List.collect (fun num -> 
        match num with
        | 0 -> [0]
        | x -> [x .. x .. upperBound - 1]) numbers
    |> List.distinct |> List.sum